mllib/src/main/scala/org/apache/spark/ml/regression/LinearRegression.scala

```diff
@@ -522,18 +522,40 @@ class LinearRegressionSummary private[regression] (
   }
 
   /**
+   * Return the residuals by type.
+   * Supported options: "deviance", "pearson", "working" and "response".
+   * Default is "deviance".
+   */
+  def residualsByType(residualsType: String = "deviance"): DataFrame = {
```

Comment:
I have not figured out a better name, ```residuals``` has been occupied. Or should we remove or deprecate  ```residuals``? @mengxr